File: cheptel_mobile/lib/data/models/constat_request.dart
class ConstatRequest {
  final int farmId;
  final int? controlSessionId;
  final String type;
  final String description;
  final double? latitude;
  final double? longitude;
  final String? localisationText;
  final String? photoUrl;
  final String? voiceMemoUrl;

  ConstatRequest({
    required this.farmId,
    this.controlSessionId,
    required this.type,
    required this.description,
    this.latitude,
    this.longitude,
    this.localisationText,
    this.photoUrl,
    this.voiceMemoUrl,
  });

  Map<String, dynamic> toJson() {
    return {
      "farmId": farmId,
      "controlSessionId": controlSessionId,
      "type": type,
      "description": description,
      "latitude": latitude,
      "longitude": longitude,
      "localisationText": localisationText,
      "photoUrl": photoUrl,
      "voiceMemoUrl": voiceMemoUrl,
    };
  }
}

File: cheptel-backend/src/main/java/com/hbtech/cheptel/dto/request/ConstatRequest.java
package com.hbtech.cheptel.dto.request;

import com.hbtech.cheptel.entity.ConstatType;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ConstatRequest {
    private Long farmId;
    private Long controlSessionId; // optionnel
    private ConstatType type;
    private String description;
    private Double latitude;
    private Double longitude;
    private String localisationText;
    private String photoUrl;
    private String voiceMemoUrl;
}